common/presenters/moves-by-to-location.js

```diff
@@ -1,17 +1,38 @@
-const { groupBy, sortBy } = require('lodash')
+const { get, groupBy, sortBy } = require('lodash')
 
+const i18n = require('../../config/i18n')
 const moveToCardComponent = require('./move-to-card-component')
 
 module.exports = function movesByToLocation(data) {
   const locations = []
   const groupedByLocation = groupBy(data, 'to_location.id')
 
-  Object.entries(groupedByLocation).forEach(([locationId, items]) => {
+  if (groupedByLocation.undefined) {
```

Comment:
Had a rethink about this with a fresh head. Take a look at the latest fixup. I decided to map the list of moves to handle setting of `to_location` there if there isn't one.

Then this function can remain the same where it always expects a `to_location` so that it can group by that.